### PR TITLE
Replace network related setup calls in kickstart.py

### PR DESCRIPTION
**Pull request by M4rtinK**
Move the network related `setup()` calls in kickstart.py, that set package requirements for the main RPM transaction and move the code to the Network DBUS module.

~~**NOTE:** This PR depends on https://github.com/rhinstaller/anaconda/pull/2073, and can be only merged once #2073 is merged. This is also the reason why the very last commit is the only relevant one. :-)~~

### CODE REVIEW

File: pyanaconda/kickstart.py [hunk 1]
```diff
@@ -289,21 +289,6 @@ def __str__(self):
...
-        super().__init__(*args, **kwargs)
-        self.packages = []
-
-    def __str__(self):
-        # The kickstart for this command is generated by the Firewall sub module
-        return ""
-
-    def setup(self):
-        firewall_proxy = NETWORK.get_proxy(FIREWALL)
-        if firewall_proxy.FirewallKickstarted:
```
Comment (poncovka) [inline pyanaconda/kickstart.py]: It looks like you can remove this DBus method.

File: pyanaconda/network.py [hunk 2]
```diff
@@ -536,15 +536,6 @@ def get_supported_devices():
     return NetworkDeviceInfo.from_structure_list(network_proxy.GetSupportedDevices())
 
 
-def get_team_devices():
```
Comment (poncovka) [inline pyanaconda/network.py]: @rvykydal, are you ok with moving the function `get_team_devices` from  `pyanaconda/network.py`?

File: tests/nosetests/pyanaconda_tests/module_network_test.py [hunk 3]
```diff
@@ -541,6 +542,50 @@ def kickstart_firewall_service_options_test(self):
...
         self._test_kickstart(ks_in, ks_out)
 
+    def default_requirements_test(self):
+        """Test that by default no packages are required by the network module."""
+        self.assertEqual(self.network_interface.CollectRequirements(), [])
+
+    def kickstart_firewall_package_requirements_test(self):
+        """Test that firewall command in kickstart results in request for firewalld package."""
+
+
```
Comment (poncovka) [inline tests/nosetests/pyanaconda_tests/module_network_test.py]: Please, remove the empty line.

File: pyanaconda/modules/network/firewall/firewall.py [hunk 4]
```diff
@@ -206,6 +207,18 @@ def set_disabled_services(self, disabled_services):
...
 
+    def collect_requirements(self):
+        """Return installation requirements for this module.
+
+        :return: a list of requirements
+        """
+        requirements = []
+
+        if self.firewall_seen:
+            requirements.append(Requirement.for_package("firewalld", reason="firewall configuration requested"))
```
Comment (poncovka) [inline pyanaconda/modules/network/firewall/firewall.py]: The attribute `reason` is being discussed at https://github.com/rhinstaller/anaconda/pull/2073#discussion_r311408871.

File: pyanaconda/kickstart.py (hunk 1 above)
Comment (M4rtinK) [reply pyanaconda/kickstart.py]: Indeed! Dropped. :)

File: tests/nosetests/pyanaconda_tests/module_network_test.py (hunk 3 above)
Comment (M4rtinK) [reply tests/nosetests/pyanaconda_tests/module_network_test.py]: Removed.

File: pyanaconda/network.py (hunk 2 above)
Comment (rvykydal) [reply pyanaconda/network.py]: Yes

**Pull request merged**